### PR TITLE
Generate webhooks with kubebuilder

### PR DESCRIPTION
**Pull request by emmjohnson**
[PR requirements]: https://github.com/vmware-tanzu/cartographer/blob/main/CONTRIBUTING.md#commit-message-and-pr-guidelines

## Changes proposed by this PR

closes # 

## Release Note

[imperative mood]: https://chris.beams.io/posts/git-commit/#imperative

## PR Checklist

Note: Please do not remove items. Mark items as done `[x]` or use ~strikethrough~ if you believe they are not relevant

- [ ] Linked to a relevant issue. Eg: `Fixes #123` or `Updates #123`
- [ ] Removed non-atomic or `wip` commits
- [ ] Filled in the [Release Note](#Release-Note) section above 
- [ ] Modified the docs to match changes

### CODE REVIEW

File: hack/overlays/webhook-configuration.yaml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+#@ load("@ytt:overlay", "overlay")
+
+#@overlay/match by=overlay.subset({"kind": "ValidatingWebhookConfiguration"})
+
```
Comment (cirocosta) [inline hack/overlays/webhook-configuration.yaml:18]: oh interesting, nice that we can do all the changes we need right here with a single overlay 👍

**Comment by cirocosta** [comment]
huh, quite a few errors that I imagine are unrelated coming from CI:

Screen Shot 2022-03-24 at 1 58 29 PM

**Review by cirocosta — APPROVED** [review]
overall, lgtm! didn't get to run locally, but looks good!

File: hack/release.sh [hunk 2]
```diff
@@ -73,6 +73,7 @@ download_ytt_to_kodata() {
 generate_release() {
         mkdir -p ./release
         ytt --ignore-unknown-comments -f ./config \
+                -f ./hack/overlays/webhook-overlay.yaml \
```
Comment (cirocosta) [inline hack/release.sh]:
```suggestion
                -f ./hack/overlays/webhook-configuration.yaml \
```

**Comment by cirocosta** [comment]
for anyone looking at this PR - note that we end up with one less `ValidatingWebhookConfiguration`

```
^                    clustersupplychainvalidator           ValidatingWebhookConfiguration  -      -   -   -
^                    deliveryvalidator                     ValidatingWebhookConfiguration  -      -   -   -
```

vs

```
^                    validating-webhook-configuration      ValidatingWebhookConfiguration  -      -   -   -
```

but, in the end, we're registering validation for the same gvrs as before (just in a single object instead).

File: config/webhook/manifests.yaml [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+    resources:
+    - clusterconfigtemplates
+  sideEffects: None
+- admissionReviewVersions:
+  - v1beta1
+  - v1
+  clientConfig:
+    service:
+      name: webhook-service
+      namespace: system
```
Comment (idoru) [inline config/webhook/manifests.yaml:49]: I noticed these changed from:
```
        name: cartographer-webhook
        namespace: cartographer-system
```

Is that intended as part of moving over to kubebuilder?

**Review by idoru — CHANGES_REQUESTED** [review]

File: config/webhook/manifests.yaml (hunk 3 above)
Comment (emmjohnson) [reply config/webhook/manifests.yaml:49]: Yes, but the overlay fixes them up

File: hack/release.sh (hunk 2 above)
Comment (emmjohnson) [reply hack/release.sh]: 😑

File: config/webhook/manifests.yaml (hunk 3 above)
Comment (idoru) [reply config/webhook/manifests.yaml:49]: Ahh, I get it now. Groovy!

**Review by idoru — APPROVED** [review]

**Pull request merged**